### PR TITLE
Fixed #24897 -- Allowed using choices longer than 1 day with DurationField

### PR DESCRIPTION
**Pull request by zauddelig**
https://code.djangoproject.com/ticket/24897

### CODE REVIEW

File: django/utils/dateparse.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
 standard_duration_re = re.compile(
     r'^'
-    r'(?:(?P<days>-?\d+) )?'
+    r'(?:(?P<days>-?\d+) (days, )?)?'
```
Comment (ewjoachim) [inline django/utils/dateparse.py]: According to the [CPython source code](https://github.com/python/cpython/blob/master/Lib/datetime.py#L455), the "s" is there only if there's more than 1 day.

**Comment by ewjoachim** [comment]
The docstring of the ```parse_duration``` [function](https://docs.djangoproject.com/en/1.8/_modules/django/utils/dateparse/#parse_duration) we're modifying says 

```python
    The preferred format for durations in Django is '%d %H:%M:%S.%f'.
```
Maybe we want to add a "days" here too...

And if it used to fail (as the issue says) and it doesn't anymore, we may want to add a regression test (which will ensure that this implementation works) ?

File: django/utils/dateparse.py (hunk 1 above)
Comment (zauddelig) [inline django/utils/dateparse.py]: Good one, fixed

**Comment by zauddelig** [comment]
Well ok, The ? indicates that the "%d %H:%M:%S.%f" format is ok and could still be the "preferred", 

Testing would surely be a good thing yet I think any test should be best fitted here:

https://github.com/django/django/tree/master/tests/forms_tests

Do you confirm?

**Comment by timgraham** [comment]
Consider a low-level test in `tests/utils_tests/test_dateparse.py`

**Comment by zauddelig** [comment]
The test were done as for the last commit (sorry for the failed tests but,  although practical, GitHub is a pain to write code in.

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
merged in 262d4db8c4c849b0fdd84550fb96472446cf90df, thanks!

**Pull request closed without merging**